sympy/stats/rv.py

```diff
@@ -1158,12 +1194,12 @@ def sample_iter_lambdify(expr, condition=None, size=(), numsamples=S.Infinity, *
     def return_generator():
         count = 0
         while count < numsamples:
-            d = ps.sample(size)  # a dictionary that maps RVs to values
+            d = ps.sample(size=size, library=library)  # a dictionary that maps RVs to values
             args = [d[rv] for rv in rvs]
 
             if condition:  # Check that these values satisfy the condition
                 gd = given_fn(*args)
-                if gd != True and gd != False:
+                if gd != True or gd != False:
```

Comment:
If I am not wrong, this will always be `True` whether, `gd` is `True`, `False` or `None`.

Reply:
```suggestion
                if gd is not True or gd is not False:
```
If I am wrong, then this can be done.

Reply:
Ohh, That was done while checking the error path, I will change that.